### PR TITLE
Feature/hotelling

### PR DESCRIPTION
**Pull request by DavidGarciaFer**
Including Hotelling's T^2 mean test for two samples.

### CODE REVIEW

File: skfda/inference/hotelling/hotelling.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+            random_state.shuffle(indices)
+            dist[i] = hotelling_t2(sample[indices[:n1]], sample[indices[n1:]],
+                                   gram)
+
+    else:  # Full permutation test
+        combinations = itertools.combinations(indices, n1)
+        dist = np.empty(int(scipy.special.comb(n, n1)))
+        for i, comb in enumerate(combinations):
+            sample1_i = np.asarray(comb)
+            sample2_i = np.setdiff1d(indices, sample1_i)
```
Comment (vnmabus) [inline skfda/inference/hotelling/hotelling.py:205]: What is this?

File: skfda/inference/hotelling/hotelling.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        k_inv = np.linalg.pinv(np.linalg.multi_dot([weights, k_pool, weights]))
+        k_inv = weights.dot(k_inv).dot(weights)
+    else:
+        # If data is discrete no weights are needed
+        k_inv = np.linalg.pinv(k_pool)
+
+    return n1 * n2 / n * m.T.dot(k_inv).dot(m)[0][0]
+
+
+def hotelling_test_ind(fd1, fd2, n_reps=None, random_state=None,
```
Comment (vnmabus) [inline skfda/inference/hotelling/hotelling.py:112]:
```suggestion
def hotelling_test_ind(fd1, fd2, *, n_reps=None, random_state=None,
```

File: skfda/inference/hotelling/hotelling.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
+from skfda.representation import FDataBasis, FData
+import numpy as np
+import itertools
+import scipy
+from sklearn.utils import check_random_state
+
+
+def hotelling_t2(fd1, fd2, weights=None):
```
Comment (vnmabus) [inline skfda/inference/hotelling/hotelling.py:8]:
```suggestion
def hotelling_t2(fd1, fd2, *, weights=None):
```

File: skfda/inference/hotelling/hotelling.py (hunk 2 above)
Comment (vnmabus) [reply skfda/inference/hotelling/hotelling.py:108]: Yeah, but here we want `n_reps` and `random_state` to be keyword-only args.

Comment (DavidGarciaFer) [reply skfda/inference/hotelling/hotelling.py]: Changed

**Pull request merged**